src/pydocstyle/config.py

```diff
@@ -208,7 +208,10 @@ def _get_config(self, node):
         if path in self._cache:
             return self._cache[path]
 
-        config_file = self._get_config_file_in_folder(path)
+        if self._options.config_file:
```

Comment:
I think I understand the confusion. In my mind, the `--config` option is meant to specify a specific config file (path to a config file), not allow a new name for config files in general. It seems more useful to me. Do you object?